Library/ViewModels/SetYourPasswordViewModel.swift

```diff
@@ -49,6 +63,32 @@ public final class SetYourPasswordViewModel: SetYourPasswordViewModelType, SetYo
       .skipRepeats()
 
     self.saveButtonIsEnabled = formIsValid
+
+    let submitFormEvent = self.saveButtonPressedProperty.signal
+
+    let saveAction = formIsValid
+      .takeWhen(submitFormEvent)
+      .filter(isTrue)
+      .ignoreValues()
+
+    let setPasswordEvent = combinedPasswords
+      .takeWhen(saveAction)
+      .map { CreatePasswordInput(password: $0.0, passwordConfirmation: $0.1) }
```

Comment:
I do this too, but when possible try to avoid the `$0.0` and `$0.1` syntax, setting variable names like `newPassword` and `confirmPassword` inside the map makes it easier to read.